src/pages/api/metrics.ts

```diff
@@ -0,0 +1,25 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { Registry } from 'prom-client';
+import { prometheus } from '../../config/prometheus';
+
+const metricsApi = async (req: NextApiRequest, res: NextApiResponse) => {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) return res.status(400).send('Bad Request');
```

Comment:
This is incredibly nitpicky, but I think it should technically be a 401 too since it's an unauthorized request.

Reply:
Not nitpicky at all, thanks for spotting!